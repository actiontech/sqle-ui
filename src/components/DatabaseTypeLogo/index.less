.database-type-logo-wrapper {
  white-space: nowrap;
  overflow: hidden;
  display: flex;
  align-items: center;

  img {
    width: 16px;
    height: 16px;
    background-color: #fff;
  }

  span {
    margin-left: 4px;
  }
}
