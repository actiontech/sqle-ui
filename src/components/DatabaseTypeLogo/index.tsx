import './index.less';
const DatabaseTypeLogo: React.FC<{ dbType: string; logoUrl?: string }> = ({
  dbType,
  logoUrl,
}) => {
  return (
    <div className="database-type-logo-wrapper">
      <img
        src={logoUrl || `/v1/static/instance_logo?instance_type=${dbType}`}
        alt=""
      />
      <span title={dbType} className="ant-select-item-option-content">
        {dbType}
      </span>
    </div>
  );
};
export default DatabaseTypeLogo;
