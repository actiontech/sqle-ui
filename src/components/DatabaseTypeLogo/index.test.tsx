import { render } from '@testing-library/react';
import DatabaseTypeLogo from '.';

describe('test DatabaseTypeLogo', () => {
  test('should match snapshot', () => {
    const { container, rerender } = render(<DatabaseTypeLogo dbType="mysql" />);

    expect(container).toMatchSnapshot();

    rerender(
      <DatabaseTypeLogo dbType="db2" logoUrl="/static/medit/logo.png" />
    );
    expect(container).toMatchSnapshot();
  });
});
