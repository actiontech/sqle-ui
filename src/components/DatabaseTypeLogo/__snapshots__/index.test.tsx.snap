// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DatabaseTypeLogo should match snapshot 1`] = `
<div>
  <div
    class="database-type-logo-wrapper"
  >
    <img
      alt=""
      src="/v1/static/instance_logo?instance_type=mysql"
    />
    <span
      class="ant-select-item-option-content"
      title="mysql"
    >
      mysql
    </span>
  </div>
</div>
`;

exports[`test DatabaseTypeLogo should match snapshot 2`] = `
<div>
  <div
    class="database-type-logo-wrapper"
  >
    <img
      alt=""
      src="/static/medit/logo.png"
    />
    <span
      class="ant-select-item-option-content"
      title="db2"
    >
      db2
    </span>
  </div>
</div>
`;
